source/bao_hyp/config.rst

```diff
@@ -3,7 +3,103 @@ Bao Configuration File
+        .shmemlist = (struct shmem[]) {
+            [0] = {/*shared memory config*/,},
+            [1] = {/*shared memory config*/,},
+            ...
+            [N] = {/*shared memory config*/,}
+        },
+
+        // Guests Configuration
+        .vmlist_size = NUM_VMs,
+        .vmlist = {
+            { /* VM 0 Config*/},
+            { /* VM 1 Config*/},
+            ...
+            { /* VM N Config*/},
+        }
+    };
+
+Before the configuration itself, it is necessary to declare the VM images using
+the ``VM_IMAGE`` macro. This macro directly embeds the guest binary file into
+the hypervisor image. Here's an example usage of the ``VM_IMAGE``:
+
+.. code-block:: c
+
+  VM_IMAGE(img_name, "/path/to/VM/binary.bin");
+
+The ``VM_IMAGE`` macro has two parameters:
+
+1. The ``img_name``, an unique identifier associated with the image that will
+   later be used to describe the image running on the VM (see `Guest Image`_);
+
+2. A C string with the guest image's binary file path. It can be either an
+   absolute path or a path relative to the config source file.
+
+The configuration takes the form of a C struct (``struct config``) defining
+name ``config``. This struct containing two distinct lists: (i) a list of
+shared memory regions (``shmemlist``) and (ii) a list of VMs (``vmlist``).
+While the list of shared memory regions is optional and may be omitted from the
+configuration, the list of VMs is mandatory and must include at least 1 VM.
+Additionally, for each list, it is necessary to specify the list size using the
+parameters ``shmemlist_size`` and ``vmlist_size``.
+
+Building Bao Hypervisor
+-----------------------
+The configuration files for the Bao hypervisor are stored in a designated
+folder known as the configuration repository - ``CONFIG_REPO``. By default,
+the ``CONFIG_REPO`` is set to the ``configs`` folder located in the top-level
+directory of the Bao hypervisor. However, users have the flexibility to specify
+a different folder by using the ``CONFIG_REPO`` option in the make command
+during the hypervisor building process. For instance, the build command of Bao
+follows the nomenclature presented below:
+
+.. code-block:: console
+
+    make -C /path/to/bao-dir\
+        PLATFORM=target-platform\
+        CONFIG_REPO=/path/to/config\
+        CONFIG=config-name\
+
+Considering a configuration named ``config-name``, the configuration naming can
+be provided in two formats:
+
+**1. Single C Source File**: Save the configuration as a C source file with the
+name ``config-name.c``.
+
+**2. Directory Format:**: Alternatively, you can organize it as a directory
+named ``config-name`` and place the C source file within, naming it
+``config.c``.
+
+In both cases, the build system will search for the configuration in the
+specified ``CONFIG_REPO`` using the defined naming convention. This allows
+users to seamlessly integrate their custom configurations into the Bao
+hypervisor build process.
+
+.. warning::
```

Comment:
This warning seems misplaced. Shouldn't it be at the end of the previous section?